File: settlingvelocity.m
function ws = settlingvelocity(rhop,rhof,g,d,nu)

Cd = @(Re)  0.25 + (24./Re) + (6./(1+Re.^(0.5)));  % drag law

Ap=(pi*d^2)/4;              % surface of the sphere - m2
Vp=(pi*d^3)/6;              % volume  of the sphere - m3

ws = fzero( @(wp) (rhop-rhof) * Vp * g ...
                   - 0.5 * Cd(wp*d/nu)* rhof * Ap * abs(wp) * wp, ...
           [1e-11 1]);

File: Python/settling_velocity.py
from scipy.optimize import least_squares
import numpy as np


def Cd(Re): return 0.4 + 24.0/Re + 6.0/(1+np.sqrt(Re))  
def Vp(d): return np.pi*d**3/6
def Ap(d): return np.pi*d**2/4

def settling_velocity(rhop=1.05, rhof=1.0 , g=9.81, d=100e-6, nu=1e-6):
    """ Settling velocity 
    
    Inputs:
        rhop : particle density, (kg/m^3)
        rhof : fluid density, (kg/m^3)
        g : gravity acceleration, default = 9.81 ( m/s^2 )
        d : particle diameter, (m)
        nu : kinematic viscosity, (m^2/s), water is 1e-6
    
    
    Outputs:
        ws : 
    
    """
    
    def f(wp): return 1e12*((rhop-rhof)*Vp(d)*g - 0.5*Cd(wp*d/nu)*rhof*Ap(d)*np.abs(wp)*wp)


    # Calculate the settling speed from the force balance in the steady-state conditions
    sol = least_squares(f, .5, bounds=[1e-11,1],ftol=1e-15,xtol=1e-8)

    return sol